src/routers/router.rs

```diff
@@ -8,89 +8,34 @@ use pyo3::types::PyAny;
 use actix_web::http::Method;
 use matchit::Router as MatchItRouter;
 
-use anyhow::{bail, Error, Result};
+use anyhow::{Context, Result};
 
-/// Contains the thread safe hashmaps of different routes
-
-pub struct Router {
-    get_routes: RwLock<MatchItRouter<(PyFunction, u8)>>,
-    post_routes: RwLock<MatchItRouter<(PyFunction, u8)>>,
-    put_routes: RwLock<MatchItRouter<(PyFunction, u8)>>,
-    delete_routes: RwLock<MatchItRouter<(PyFunction, u8)>>,
-    patch_routes: RwLock<MatchItRouter<(PyFunction, u8)>>,
-    head_routes: RwLock<MatchItRouter<(PyFunction, u8)>>,
-    options_routes: RwLock<MatchItRouter<(PyFunction, u8)>>,
-    connect_routes: RwLock<MatchItRouter<(PyFunction, u8)>>,
-    trace_routes: RwLock<MatchItRouter<(PyFunction, u8)>>,
-}
-
-impl Router {
-    pub fn new() -> Self {
-        Self {
-            get_routes: RwLock::new(MatchItRouter::new()),
-            post_routes: RwLock::new(MatchItRouter::new()),
-            put_routes: RwLock::new(MatchItRouter::new()),
-            delete_routes: RwLock::new(MatchItRouter::new()),
-            patch_routes: RwLock::new(MatchItRouter::new()),
-            head_routes: RwLock::new(MatchItRouter::new()),
-            options_routes: RwLock::new(MatchItRouter::new()),
-            connect_routes: RwLock::new(MatchItRouter::new()),
-            trace_routes: RwLock::new(MatchItRouter::new()),
-        }
-    }
-
-    #[inline]
-    fn get_relevant_map(&self, route: Method) -> Option<&RwLock<MatchItRouter<(PyFunction, u8)>>> {
-        match route {
-            Method::GET => Some(&self.get_routes),
-            Method::POST => Some(&self.post_routes),
-            Method::PUT => Some(&self.put_routes),
-            Method::PATCH => Some(&self.patch_routes),
-            Method::DELETE => Some(&self.delete_routes),
-            Method::HEAD => Some(&self.head_routes),
-            Method::OPTIONS => Some(&self.options_routes),
-            Method::CONNECT => Some(&self.connect_routes),
-            Method::TRACE => Some(&self.trace_routes),
-            _ => None,
-        }
-    }
+use super::{RouteType, Router};
 
-    #[inline]
-    fn get_relevant_map_str(
-        &self,
-        route: &str,
-    ) -> Option<&RwLock<MatchItRouter<(PyFunction, u8)>>> {
-        if route != "WS" {
-            let method = match Method::from_bytes(route.as_bytes()) {
-                Ok(res) => res,
-                Err(_) => return None,
-            };
+type RouteMap = RwLock<MatchItRouter<(PyFunction, u8)>>;
 
-            self.get_relevant_map(method)
-        } else {
-            None
-        }
-    }
+/// Contains the thread safe hashmaps of different routes
+pub struct DynRouter {
```

Comment:
Is `DynRouter` a `HTTPRouter`?

Reply:
What do you mean by that?
This is the struct that was previously called `Router`. To my understanding it stores the routes that have parameters, and I called it `DynRouter` to oppose it to `ConstRouter`. I'm not familiar with routers so I may have not understood everything, let me know if you prefer me to rename it

Reply:
@AntoineRR , the vanilla `Router`(now `DynRouter`) just serves HTTP Requests. I am not sure what you mean by `DynRouter`. I was thinking of something like `HTTPRouter` or `HTTPRequestRouter`.

What do you think?

Reply:
It makes sense! I renamed it to `HttpRouter` and changed the file name as well.